viewer/src/main/java/nl/b3p/viewer/stripes/DownloadFeaturesActionBean.java

```diff
@@ -294,6 +308,23 @@ public void stream(HttpServletResponse response) throws Exception {
         }
     }
 
+    private List<ConfiguredAttribute> filterAttributes(List<ConfiguredAttribute> attributes) {
+        List<ConfiguredAttribute> newAttrs = new ArrayList<>();
+        if (!columns.isEmpty()) {
+            for (String column : columns) {
+                for (ConfiguredAttribute attr : attributes) {
```

Comment:
chances are that `attributes` is always larger (and has larger objects) than `columns` so having the outer loop over `attributes` might be faster I think, as well as using a filtered stream.

Reply:
also, after switching around, you can stop once `newAttrs.size()` reaches `columns.size()` as there won't be more to add

Reply:
Funny, was actually optimising for that (see also: https://stackoverflow.com/questions/35710346/why-is-there-a-performance-difference-between-the-order-of-a-nested-loop). Because of the break on line 318, it would be faster to have the smaller array (columns) on the outside.

By inference, newAttrs.size() can never be larger than columns.size()